cudax/include/cuda/experimental/__stf/internal/cuda_kernel_scope.cuh

```diff
@@ -37,6 +37,28 @@ namespace cuda::experimental::stf
 class graph_ctx;
 class stream_ctx;
 
+namespace reserved
+{
+
+template <typename T>
+struct is_function_or_kernel : ::std::false_type
+{};
+
+template <>
+struct is_function_or_kernel<CUfunction> : ::std::true_type
+{};
+
+#if CUDA_VERSION >= 12000
```

Comment:
Important: Please use either `_CCCL_CTK_AT_LEAST(12, 0)` or any of the compiler version check macros, depending on when `CUkernel` is available. Does it depend on the CTK version or nvcc version? And consider whether it handles clang in CUDA mode correctly.